src/engine/specification.md

```diff
@@ -63,22 +63,22 @@ This structure contains the attributes (in addition to the other parameters cont
   - `VALID` - given payload is valid
   - `INVALID` - given payload is invalid
   - `SYNCING` - sync process is in progress
+2. `validAncestorHash`: `DATA|None`, 32 Bytes - block hash of the most recent *valid* ancestor of the payload
 
 #### Specification
 
-1. Client software **MUST** validate the payload according to the execution environment rule set with modifications to this rule set defined in the [Block Validity](https://eips.ethereum.org/EIPS/eip-3675#block-validity) section of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) and respond with the validation result.
+1. Client software **MUST** validate the payload according to the execution environment rule set with modifications to this rule set defined in the [Block Validity](https://eips.ethereum.org/EIPS/eip-3675#block-validity) section of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) and return the validation result.
+    * If validation succeeds, return `{status: VALID, validAncestorHash: payload.blockHash}`
+    * If validation fails, return `{status: INVALID, validAncestorHash: ancestorHash}` where `ancestorHash` is the block hash of the most recent *valid* ancestor of the invalid payload. That is, the ancestor of the payload in the valid block tree with the highest `blockNumber`.
```

Comment:
There are really only a couple of cases here. 
1. We have the parent, the parent is fine, but this block is bad. We return parentHash, 
2. We have the parent, we tried to import it earlier, but it did not pass validation. We follow that 'bad' sidechain up to the common canon ancestor, return that block, 

In the second case, is it expected that the EL stores/tracks invalid sidechains? Not sure if that has been expressed in the spec excplicitly?

Reply:
One viable alternative would be for upon receiving `INVALID`, CL to go to the earliest `SYNCING` ancestor and walk up the tree calling `executePayload` marking things as `VALID` until they find the first `INVALID` ancestor